tests/testthat/test-aorsf_data.R

```diff
@@ -0,0 +1,176 @@
+library(testthat)
+
+mtcars_orsf <- mtcars
+mtcars_orsf$vs <- factor(mtcars_orsf$vs)
+mtcars_na <- mtcars_orsf
+mtcars_na$cyl[1] <- NA
+
+
+test_that("regression model object", {
+
+  skip_if_not_installed("aorsf", "0.1.3")
+
+  set.seed(1234)
+  aorsf_regr_fit <- aorsf::orsf(
+    # everyone's favorite
+    data = mtcars_orsf, formula = mpg ~ .,
+    # faster
+    n_tree = 10,
+    # requested default from tidymodels
+    n_thread = 1
+  )
+
+  # formula method
+  regr_spec <- rand_forest(trees = 10) %>%
+    set_engine("aorsf") %>%
+    set_mode("regression")
+
+  # not sure if the expect_error is needed, but copying from censored
+  set.seed(1234)
+  expect_error(
+    bonsai_regr_fit <- fit(regr_spec, data = mtcars_orsf, mpg ~ .),
+    NA
+  )
+
+  expect_equal(
+    bonsai_regr_fit$fit,
+    aorsf_regr_fit,
+    ignore_formula_env = TRUE
+  )
+
+})
+
+test_that("classification model object", {
+
+  skip_if_not_installed("aorsf", "0.1.3")
+
+  set.seed(1234)
+  aorsf_clsf_fit <- aorsf::orsf(
+    data = mtcars_orsf, formula = vs ~ .,
+    n_tree = 10,
+    n_thread = 1
+  )
+
+  # formula method
+  clsf_spec <- rand_forest(trees = 10) %>%
+    set_engine("aorsf") %>%
+    set_mode("classification")
+
+  set.seed(1234)
+  expect_error(
+    bonsai_clsf_fit <- fit(clsf_spec, data = mtcars_orsf, vs ~ .),
+    NA
+  )
+
+  expect_equal(
+    bonsai_clsf_fit$fit,
+    aorsf_clsf_fit,
+    ignore_formula_env = TRUE
+  )
+
+})
+
+test_that("regression predictions", {
+
+  # bug-fix for na_action = 'pass' was pushed in this version
+  skip_if_not_installed("aorsf", "0.1.4.9001")
+
+  set.seed(1234)
+  aorsf_regr_fit <- aorsf::orsf(
+    data = mtcars_orsf, formula = mpg ~ .,
+    n_tree = 10
+  )
+
+  aorsf_regr_pred <- predict(
+    aorsf_regr_fit,
+    new_data = mtcars_na,
+    na_action = 'pass'
+  )
+
+  # formula method
+  regr_spec <- rand_forest(trees = 10) %>%
+    set_engine("aorsf") %>%
+    set_mode("regression")
+
+  set.seed(1234)
+  bonsai_regr_fit <- fit(regr_spec, mpg ~ ., data = mtcars_orsf)
+  bonsai_regr_pred <- predict(bonsai_regr_fit, new_data = mtcars_na)
+
+  expect_s3_class(bonsai_regr_pred, "tbl_df")
+  expect_true(all(names(bonsai_regr_pred) == ".pred"))
+  expect_equal(bonsai_regr_pred$.pred, as.vector(aorsf_regr_pred))
+  expect_equal(nrow(bonsai_regr_pred), nrow(mtcars_orsf))
+
+  # single observation
+  pred_1row <- predict(bonsai_regr_fit, mtcars_orsf[2,])
+  expect_identical(nrow(pred_1row), 1L)
+
+})
+
+test_that("classification predictions", {
+
+  # bug-fix for na_action = 'pass' was pushed in this version
+  skip_if_not_installed("aorsf", "0.1.4.9001")
```

Comment:
Out of curiosity, when do you anticipate this release will be out?

Reply:
I think CRAN prefers at most 1 update per month, so early June would be a good time since 0.1.4 was submitted around May 1.

Reply:
I think I'd be hesitant to send these changes out before those NA action changes are on CRAN. Was starting to gear up for a bonsai release but can sit tight for a couple more weeks so we can add aorsf support in that one!

Reply:
Totally makes sense. No rush on my end. =] It's my bad for not checking the classification and regression modules in `aorsf` more carefully before I released 0.1.4.